src/computability/language.lean

```diff
@@ -151,4 +151,98 @@ begin
     finish }
 end
 
+lemma le_iff (l m : language α) : l ≤ m ↔ l + m = m :=
+begin
+  rw [set.le_eq_subset, add_def],
+  split,
+  { intro h,
+    ext x,
+    simp only [set.mem_union_eq, or_iff_right_iff_imp],
+    tauto },
+  { intro h,
+    rw ←h,
+    exact set.subset_union_left l m }
+end
+
+lemma one_add_lang_lang_star_le_lang_star (l : language α) : 1 + l*l.star ≤ l.star :=
+begin
+  rw [set.le_eq_subset, one_def, add_def, mul_def, star_def],
+  intros x hx,
+  simp only [exists_and_distrib_left, set.mem_image2, set.image_prod, set.mem_union_eq,
+    set.mem_set_of_eq] at hx,
+  cases hx,
+  { rw [set.mem_singleton_iff] at hx,
+    subst hx,
+    use [[]],
+    split,
+    refl,
+    tauto },
+  { rcases hx with ⟨ a, ha, b, ⟨ ⟨ S, hb, hS ⟩, hx ⟩ ⟩,
+    use a :: S,
+    split,
+    finish,
+    rintros y (hy|hy);
+    finish }
+end
+
+lemma one_add_lang_star_lang_le_lang_star (l : language α) : 1 + l.star*l ≤ l.star :=
```

Comment:
```suggestion
lemma one_add_star_mul_self_le_star (l : language α) : 1 + l.star*l ≤ l.star :=
```

Reply:
Can you prove that `l.star*l = l*l.star`, which would make this proof follow trivially from the previous one?